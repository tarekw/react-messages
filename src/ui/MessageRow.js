import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const MessageRow = ({ messageItem={}, onMouseEnter=f=>f, showEmail=false }) => (
  <tr onMouseEnter={() => onMouseEnter(messageItem.id)}>
    <td>
      <img
        className="img-rounded"
        src={messageItem.imgSrc}
        style={{width:100, height:100}}
        alt=""
      />
    </td>
    <td>
      {
        showEmail ?
          <div>{messageItem.userEmail}</div>
        :
          <div>
            <div>{messageItem.message}</div>
            <div>{moment(messageItem.timestamp).format('MMMM Do YYYY, h:mm:ss a')}</div>
          </div>
      }
    </td>
  </tr>
);

MessageRow.propTypes = {
    messageItem: PropTypes.object
}

export default MessageRow
