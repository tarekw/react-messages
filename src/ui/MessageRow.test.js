import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import MessageRow from './MessageRow';

jest.mock('moment', () => () => ({format: () => '2019–09–13T12:00:00+00:00'}));

it('renders correctly', () => {
  const props = {
    messagesItem: {},
  };
  const tree = shallow(<MessageRow {...props} />);
  expect(tree).toMatchSnapshot();
});

it('should handle mouse enter event', () => {
  const props = {
    messageItem: {
        id: '1',
    },
    onMouseEnter: jest.fn(),
  };

  const tree = renderer.create(<MessageRow {...props} />);

  const trTag = tree.root.findAllByType('tr')[0];

  trTag.props.onMouseEnter();

  expect(props.onMouseEnter).toHaveBeenCalled();
  expect(props.onMouseEnter).toHaveBeenCalledWith(props.messageItem.id);
});
