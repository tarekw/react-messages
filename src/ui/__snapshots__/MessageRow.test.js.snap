// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  onMouseEnter={[Function]}
>
  <td>
    <img
      alt=""
      className="img-rounded"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </td>
  <td>
    <div>
      <div />
      <div>
        2019–09–13T12:00:00+00:00
      </div>
    </div>
  </td>
</tr>
`;
